src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+                        }
+			String response = msg.getResponseBody().toString();
+			System.out.println("Letter : " + letter + "Response : " + response);
+			Pattern pattern = Pattern.compile(regex.getText());
+			Matcher matcher = pattern.matcher(response);
+			if (matcher.find()) { 
+				model.addRow(new Object[]{letter, "FAILED"});
+				System.out.println("Match Found"); 
+				i_chars.append(letter).append(" \n");
+			}
+			else {
+				model.addRow(new Object[]{letter, "SUCCESS"});
+			}
+//msg.getResponseHeader().getStatusCode() + " " + msg.getResponseHeader().getReasonPhrase()); 
+			persistAndShowMessage(msg);
+			}
+			}
+
+		});
+		blacklist.addActionListener(new ActionListener () {
+		@Override
+			public void actionPerformed(ActionEvent event) {
+				JOptionPane.showMessageDialog(FieldEnumeration.this,"Illegal Characters: \n" + i_chars.toString());	
+			}
+                });
+
+		pack();
+			
+		return Panel;
+	}
+	public HistoryReference getHistoryRef() {
+                return historyRef;
+        }
+
+	private HttpSender gethttpSender() {
+        if (httpSender == null) {
+            httpSender = new HttpSender(Model.getSingleton().getOptionsParam()
+                    .getConnectionParam(), false, HttpSender.MANUAL_REQUEST_INITIATOR);
+        }
+        	return httpSender;
+    	}
+
+        public void setHistoryRef(HistoryReference historyRef) {
+                this.historyRef = historyRef;
+		StringBuilder sb = new StringBuilder();
+                sb.append(historyRef.getURI().toString());
+		printURL.setText(sb.toString());
+		c1.gridx = 0;
+                c1.gridy = 5;
+		JLabel l5 = new JLabel("Form Parameters: ");
+		l5.setFont(font);
+		jPanel.add(l5, c1);
+		formF.removeAll();
+		formF.setLayout(new GridBagLayout());
+		GridBagConstraints gbcForm = new GridBagConstraints();
+		gbcForm.gridy = 0;
+		gbcForm.anchor = GridBagConstraints.NORTH;
+
+		try {
+		System.out.println("Entered Params");
+		listParam = Model.getSingleton().getSession().getParameters(historyRef.getHttpMessage(), HtmlParameter.Type.form);
+		for (org.zaproxy.zap.model.NameValuePair parameter : listParam) {
+			System.out.println("Form Params: ");
+                        System.out.println(parameter.getName()+" :: "+parameter.getValue());
+			JRadioButton button = new JRadioButton(parameter.getName());
+			group.add(button);
+			gbcForm.gridx = 0;
+		    	gbcForm.fill = GridBagConstraints.NONE;
+			formF.add(button, gbcForm);
+			params.addItem(parameter.getName());
+
+			ZapTextField textField = new ZapTextField(parameter.getValue());
+    			textField.setColumns(10);
+ 
+    			gbcForm.gridx = 1;
+			gbcForm.fill = GridBagConstraints.HORIZONTAL;
+	    		formF.add(textField, gbcForm);
+ 
+    			gbcForm.gridy++;
+                }
+		c1.gridx = 0;
+                c1.gridy = 6;
+		jPanel.add(params, c1);
+		c1.gridx = 0;
+                c1.gridy = 7;
+		JLabel prefix = new JLabel("prefix");
+		prefix.setFont(font);
+		jPanel.add(prefix, c1);
```

Comment:
When the Field Enumeration dialogue is opened again it adds a new label (leading to twice shown).

Reply:
Yes.. That is the main issue. I am not sure, why does this happen?

Reply:
The labels are being added to the `jPanel` each time the dialogue is invoked, we need to remove all components or, better, just add the components once and just change the values of the fields when the `HistoryReference` is set.